### PR TITLE
App.start_loop() context manager

### PR DESCRIPTION
**Pull request by jamesdbrock**
Also:

Assorted type annotations.

StateValue test_subscribe_unmount

### CODE REVIEW

File: edifice/app.py [hunk 1]
```diff
@@ -253,7 +254,7 @@ def _make_widget_helper(comp):
         return _make_widget_helper(self._root)
 
     def start(self):
-        """Start the QEventLoop.
+        """Start the application event loop.
```
Comment (considerate) [inline edifice/app.py:257]: I think this should be implemented in terms of `start_loop`.

File: edifice/app.py [hunk 2]
```diff
@@ -276,3 +277,54 @@ def cleanup(e):
...
+
+                self._inspector_component = inspector.Inspector(
+                    self._render_engine._component_tree, self._root,
+                    refresh=(lambda: (self._render_engine._component_tree, self._root)))
+                icon_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), "inspector/icon.png")
+                component = Window(title="Component Inspector", on_close=cleanup, icon=icon_path)(self._inspector_component)
+                component._edifice_internal_parent = None
+                self._request_rerender([component], {})
+            yield loop
+            loop.close()
```
Comment (considerate) [inline edifice/app.py:327]:
```suggestion
            loop = QEventLoop(self.app)
            try:
                asyncio.set_event_loop(loop)
                self._request_rerender([self._root], {})
                if self._inspector:
                    logger.info("Running inspector")
                    def cleanup(e):
                        self._inspector_component = None

                    self._inspector_component = inspector.Inspector(
                        self._render_engine._component_tree, self._root,
                        refresh=(lambda: (self._render_engine._component_tree, self._root)))
                    icon_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), "inspector/icon.png")
                    component = Window(title="Component Inspector", on_close=cleanup, icon=icon_path)(self._inspector_component)
                    component._edifice_internal_parent = None
                    self._request_rerender([component], {})
                yield loop
            finally:
                loop.close()

```

**Review by considerate — CHANGES_REQUESTED** [review]
Looks good overall, I think the following should be changed before we merge, though.

File: edifice/app.py (hunk 1 above)
Comment (jamesdbrock) [reply edifice/app.py]: Done.

File: edifice/app.py (hunk 2 above)
Comment (jamesdbrock) [reply edifice/app.py]: Done.

**Pull request merged**